backend/corpora/common/auth0_manager.py

```diff
@@ -0,0 +1,128 @@
+import requests
+
+from backend.corpora.common.corpora_config import CorporaAuthConfig
+from backend.corpora.common.utils import json
+
+
+class Auth0ManagementSession:
+    """
+    A wrapper around the Auth0 Management API. https://auth0.com/docs/api/management/v2
+    """
+    _session = None
+    _domain = None
+
+    @property
+    def session(self):
+        if not self._session:
+            domain = self.domain
+            session = requests.Session()
+
+            def _refresh_token(r, *args, **kwargs):
+                """Automatically refresh the auth0 management token if it expires."""
+                if r.status_code == 401:
+                    token = self.get_auth0_management_token(domain)
+                    session.headers.update({"Authorization": token})
+                    r.request.headers["Authorization"] = session.headers["Authorization"]
+                    return session.send(r.request)
+
+            session.hooks["response"].append(_refresh_token)
+            self._session = session
+        return self._session
+
+    @property
+    def domain(self):
+        if not self._domain:
+            self._domain = CorporaAuthConfig.domain or "test"
+        return self._domain
+
+    @domain.setter
+    def domain(self, domain: str):
+        self._domain = domain
+
+    def __getattr__(self, item):
+        return getattr(self.session, item)
```

Comment:
how does this differ from `__getattribute__` as is used by the `Entity` class in the other data portal code